### PR TITLE
Remove Id Usages from LineageSerializer

### PR DESCRIPTION
**Pull request by knazir**
Missed in https://github.com/caskdata/cdap/pull/6719

### CODE REVIEW

**Comment by knazir** [comment]
Build running: http://builds.cask.co/browse/CDAP-DUT4813-1

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/lineage/LineageSerializer.java [hunk 1]
```diff
@@ -39,11 +42,11 @@
  */
 // TODO: Clean up this class (make data/program keys) when new Id classes are used CDAP-4291
```
Comment (poornachandra) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/lineage/LineageSerializer.java]: Can we handle this TODO also?

**Review by poornachandra — COMMENTED** [review]
@knazir Changes LGTM, just one question.

**Comment by knazir** [comment]
@poornachandra talked to @bdmogal about it and it was decided that the TODO is not needed.

**Comment by knazir** [comment]
Guild was breen: http://builds.cask.co/browse/CDAP-DUT4813-1

**Pull request merged**